### PR TITLE
perform refactoring of duplicated navigation code

### LINKED ISSUE #169: Refactor duplicated navigation related code
The following code appears in many files and could be refactored:

    val callback = object : OnBackPressedCallback(true) {
        override fun handleOnBackPressed() {
            findNavController().popBackStack()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        callback.isEnabled = false
        callback.remove()
        _binding = null
    }

### PR DESCRIPTION
**Pull request by martin-lenweiter**
Actually, since all the callbacks were just popping the backstack, I just deleted all of them and let the navigation component handle this.

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/NavigationUtils.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.github.HumanLearning2021.HumanLearningApp.view
+
+import androidx.activity.OnBackPressedCallback
+import androidx.navigation.NavController
+
+class NavigationUtils {
+    companion object {
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/NavigationUtils.kt]:
```suggestion
object NavigationUtils {
```

We will only ever have the object, just like `TestUtils`

**Review by NicolasVial — APPROVED** [review]
Good cleanup :) looks good to me !

**Review by Nitwix — APPROVED** [review]
Excellent! I'm happy that you could remove all this boilerplate

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureActivityTest.kt [hunk 2]
```diff
@@ -72,12 +70,6 @@ class AddPictureActivityTest {
         BaristaVisibilityAssertions.assertDisplayed(R.id.use_camera)
     }
 
-    @Test
-    fun backButtonWorks() {
```
Comment (bbjubjub2494) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureActivityTest.kt:76]: do we know for a fact that the functionality is ok?

Comment (martin-lenweiter) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureActivityTest.kt:76]: Only from manual testing. I'll add a couple navigation tests to make sure.

Comment (martin-lenweiter) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureActivityTest.kt:76]: Actually there already are some tests in GeneralNavigationTest, so I added only one more.

Comment (bbjubjub2494) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureActivityTest.kt:76]: ok I'm happy

**Review by bbjubjub2494 — APPROVED** [review]

**Pull request merged**